ui/core-react/src/core-react/inputs/numberinput/NumberInput.tsx

```diff
@@ -294,7 +295,7 @@ const ForwardRefNumberInput = React.forwardRef<
         onKeyDown={handleKeyDown}
         onFocus={handleFocus}
         onBlur={handleBlur}
-        size="small"
+        size={size}
```

Comment:
This one is confusing, as we were actually defaulting to `"small"` if nothing was provided, and `"medium"` was not an option (as `"small"` and `undefined` would both be `"medium"`, and `"large"` would be the only thing doing something.)

For this one in particular, add the full options list: `"small" | "medium" | "large"` and keep `undefined` as `small` so the behavior do not change.

Reply:
The InputProps from iTwinUI that the NumberInput is extending only allows for "small", "large", or undefined - undefined implying the default "medium" size. Having undefined also be "small" in this instance I think would prevent the "medium" size from being usable

Reply:
I know, that's why I ask to override the ItwinUI props that we use with a full set `"small" "medium" "large"` and change the default (`undefined` to mean small like it used to be.)